sorting/external_sort.cpp

```diff
@@ -0,0 +1,187 @@
+            string temp_file_name = to_string(chunk_file_names.size());
+            ofstream data_file(temp_file_name, ofstream::trunc);
+            for (int val : chunk) {
+                data_file << val << "\n";
+            }
+            data_file.close();
+            chunk_file_names.push_back(temp_file_name);
+
+            chunk.clear();
+            line_number = 0;
+        }
+    }
+    // sort the last chunk
+    if (line_number != 0) {
+        sort(chunk.begin(), chunk.end());
+
+        // write sort result to temporary file. use index as file name
+        string temp_file_name = to_string(chunk_file_names.size());
+        ofstream data_file(temp_file_name, ofstream::trunc);
+        for (int val : chunk) {
+            data_file << val << "\n";
+        }
+        data_file.close();
+        chunk_file_names.push_back(temp_file_name);
+    }
+
+    cout << "End sort chunks" << endl;
+    cout << "Merge sorted chunks using k-way merging" << endl;
+
+    // merge all sorted chunks to the final result, using K-Way merging
+    vector<ChunkFile*> chunk_files;
+
+    // change to absolutely path if u want
+    ofstream result_file("sort_result.txt", ofstream::trunc);
+
+    // open chunk files and read the first value
+    for (string f_name : chunk_file_names) {
+        ChunkFile* tmp_chunk_file = new ChunkFile();
+        tmp_chunk_file->file = ifstream(f_name);
+        getline(tmp_chunk_file->file, tmp_chunk_file->line);
+        tmp_chunk_file->currentVal = stoi(tmp_chunk_file->line);
+        chunk_files.push_back(tmp_chunk_file);
+    }
+
+    while (true) {
+        // find min value (asc sort)
+        int min_index = 0;
+
+        for (int i = 1; i < chunk_files.size(); i++) {
+            if (chunk_files[min_index]->currentVal >
+                chunk_files[i]->currentVal) {
+                min_index = i;
+            }
+        }
+        // write min value to result
+        result_file << chunk_files[min_index]->currentVal << "\n";
+        // read new value
+        if (getline(chunk_files[min_index]->file,
+                    chunk_files[min_index]->line)) {
+            chunk_files[min_index]->currentVal =
+                stoi(chunk_files[min_index]->line);
+        } else {
+            delete chunk_files[min_index];
+            chunk_files.erase(chunk_files.begin() + min_index);
+        }
+
+        // if all files are read, then stop
+        if (chunk_files.size() == 0)
+            break;
+    }
+
+    cout << "End merge sorted chunks" << endl;
+    result_file.close();
+
+    // delete temporary files
+    for (string temp_file : chunk_file_names) {
+        remove(temp_file.c_str());
+    }
+}
+
+/**
+ * Test the sort result. This method only works when the mock data fit in
+ * memory. (in most real cases, it doesn't)
+ */
+void test() {
+    cout << "Testing" << endl;
+
+    string line;
+
+    // read all mock data
+    vector<int> mock_data;
```

Comment:
```suggestion
    vector<int> mock_data{};
```
initialize your variables